cpp/src/arrow/array/validate.cc

```diff
@@ -699,57 +713,147 @@ struct ValidateArrayImpl {
     return Status::OK();
   }
 
+ private:
+  /// \pre basic validation has already been performed
+  template <typename offset_type>
+  Status FullyValidateOffsets(int64_t offset_limit) {
```

Comment:
So this is for List rather than ListView?

Reply:
Yes. The ListView function is `FullyValidateOffsetsAndValues` as they have to be validated together to guarantee `offset[i] + size[i] <= values.length()`.